src/cargo/util/context/mod.rs

```diff
@@ -3010,4 +3015,37 @@ mod tests {
             assert_eq!(http.multiplexing, result);
         }
     }
+
+    #[test]
+    fn env_argument_parsing() {
+        let mut gctx = GlobalContext::new(Shell::new(), "".into(), "".into());
+        gctx.set_search_stop_path(std::path::PathBuf::new());
+
+        let values: &[(&str, &[&str])] = &[
+            ("VAL1", &["--path"]),
+            ("VAL2", &["--path=\"y z\""]),
+            ("VAL3", &["--path", "\"y z\""]),
+        ];
+
+        let mut env = HashMap::new();
+        for (key, value) in values {
+            env.insert(format!("CARGO_{key}"), value.join(" "));
+        }
+        gctx.set_env(env);
+
+        for (key, value) in values {
+            let mut args = Vec::new();
+            gctx.get_env_list(&ConfigKey::from_str(key), &mut args)
+                .unwrap();
+
+            let mut expected = Vec::new();
+            for sub in *value {
+                expected.push((
+                    sub.replace("\"", ""),
```

Comment:
`shlex` removes (double) quotes. Not sure whether or not it's a good thing. If it's not the expected behaviour, I can implement a simple shell parser which keeps them.